src/metabase/models/notification.clj

```diff
@@ -0,0 +1,56 @@
+(ns metabase.models.notification
+  (:require [metabase.util :as u]
+            [metabase.util.schema :as su]
+            [schema.core :as s]
+            [toucan.db :as db]
+            [toucan.models :as models]))
+
+(def notifier-types
+  "Schema enum of the acceptable values for the `notifier_type` column"
+  (s/enum "comment" "moderation_review"))
+
+(def notifier->model
+  "Maps DB name of notifier to the model symbol (used for db/select and such)"
+  {"comment"           'Comment
+   "moderation_review" 'ModerationReview})
+
+(models/defmodel Notification :notification)
+(u/strict-extend (class Notification)
+  models/IModel
+  (merge models/IModelDefaults
+         {:properties (constantly {:timestamped? true})}))
+
+(defn- in-clause
+  [ids]
+  (if ids
+    [:in :id (distinct ids)]
+    [:= 0 1]))
```

Comment:
I don't understand this clause?

Reply:
It barfs if `ids` is empty:

```
(db/select 'ModerationReview {:where [:in :id (distinct [])]})

Execution error (PSQLException) at org.postgresql.core.v3.QueryExecutorImpl/receiveErrorResponse (QueryExecutorImpl.java:2553).
ERROR: syntax error at or near ")"
  Position: 51
  ```
  
  so replacing the `in` clause with `[:= 0 1]` guarantees it'll evaluate to false